### ISSUE TITLE
Do not apply roles if none is specified to get

### ISSUE THREAD

**Issue by hendrul** [mickhansen/ssacl-attribute-roles#9]
I need to restrict the visibility of some attributes to some roles, only when sending data to client, but I still need to access this properties from the instance using the readable "instance.attrib" syntax. So reading the documentation I see that with set() if you don't specify any role, by default it apply "self".

``` javascript
var user = sequelize.define('user', {
  ...
  rank: {
     type: Sequelize.STRING,
     roles: {
       self: {set: false, get: true}
       admin: true
     }
  }
}
user.set({rank: 'UBER'}, {role: 'self'||undefined}); // Will be ignored
```

So I thought the same apply for get(), and with this role definition:

``` javascript
password: {
  roles: {
    self: true,
    client: false
  }
}
```

I could achieve what I want

``` javascript
//getter with no roles defined defaults to 'self'
user.password //123
//when returning to client:
user.get({role: 'client'}) //password is hidden
```

In short, I want to be able to only hide attributes when I need it

**Comment by mickhansen** [comment]
We don't actually apply any default role, `self` is just an example.

**Comment by hendrul** [comment]
I think it would be a good idea, because showing all attrs when not asking for any role in particular, could help me solve two issues I'm having right now: 
1. using "instance.attrib" syntax to access all fields, and only when I need to hide attributes I specify roles.
2. using association accessors fails when the foreign key has some role defined

Maybe we could setup this behavior for get() and set() through config

**Comment by hendrul** [comment]
By the way, there is no support for ACLs with set() as the documentation says, only get is overwritten.

**Comment by mickhansen** [comment]
@hendrul intent was set aswell, but never got around to it.

**Comment by hendrul** [comment]
In my case I need to show all attributes by default when instance.get() but apply ACLs by default when instance.set() (well, when set() functionality gets implemented) the behavior for both should be configurable.

**Comment by drunkhacker** [comment]
this is super useful if implemented. Actually I think it can be done with quick fix like changing 
`if (!attr || !attr.roles || attr.roles && attr.roles[options.role] && attr.roles[options.role].get) {` to 
`if (!attr || !attr.roles || !options.role || attr.roles && attr.roles[options.role] && attr.roles[options.role].get) {` in lib/index.js

**Comment by mickhansen** [comment]
@drunkhacker happy to review a PR but not actively maintaining this project anymore.
